test/suite/test_reconcile01.py

```diff
@@ -0,0 +1,139 @@
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+from wtdataset import SimpleDataSet
+
+# test_reconcile01.py
+#
+# Test scenarios of removing a non existing key/recno leading to implict delete
+# records only to the fixed length column store. Performing eviction on the page
+# when they are globally visible or not and expecting them to read back as 0.
+class test_reconcile01(wttest.WiredTigerTestCase):
+    conn_config = 'in_memory=false'
+
+    format_values = [
+        ('column', dict(key_format='r', value_format='S')),
+        ('column-fix', dict(key_format='r', value_format='8t')),
+        ('integer-row', dict(key_format='i', value_format='S')),
+    ]
+
+    long_running_txn_values = [
+       ('long-running', dict(long_run_txn=True)),
+       ('no-long-running', dict(long_run_txn=False))
+    ]
+
+    scenarios = make_scenarios(format_values, long_running_txn_values)
+
+    # Evict the page to force reconciliation.
+    def evict(self, ds, uri, key, check_value):
+        evict_cursor = ds.open_cursor(uri, None, "debug=(release_evict)")
+        self.session.begin_transaction()
+        v = evict_cursor[key]
+        self.assertEqual(v, check_value)
+        self.assertEqual(evict_cursor.reset(), 0)
+        self.session.rollback_transaction()
+        evict_cursor.close()
+
+    @wttest.skip_for_hook("timestamp", "crashes in evict function, during cursor reset")  # FIXME-WT-9809
+    def test_flcs(self):
+        uri = "table:test_reconcile01"
+        nrows = 44
+        ds = SimpleDataSet(
+            self, uri, nrows, key_format=self.key_format, value_format=self.value_format, config='leaf_page_max=4096')
+        ds.populate()
+
+        appendkey1 = nrows + 10
+        appendkey2 = nrows + 17
+
+        # Write a few records.
+        cursor = ds.open_cursor(uri)
+        self.session.begin_transaction()
+        for i in range(1, nrows + 1):
+            if self.value_format == '8t':
+                cursor[i] = i
+            else:
+                cursor[i] = str(i)
+        self.session.commit_transaction()
+
+        # Start a long running transaction.
```

Comment:
I could not make the test fail when long-running txn is disabled, how come it is related?

Reply:
Yes, the test fails only when the implicit tombstone is not globally visible. I added this scenario just to make sure that we behave the same when it is globally visible also.